BootloaderCommonPkg/Include/Library/CryptoLib.h

```diff
@@ -52,10 +58,31 @@ typedef UINT8 HASH_CTX[IPP_HASH_CTX_SIZE];   //IPP Hash context buffer
 // RSA verify required key parameter format
 //
 typedef struct {
-  UINT32                   Signature;
-  UINT8                    PubKeyData[0];
+  UINT32                   PubExp;
+  UINT8                    Modulus[0];
 } RSA_PUB_KEY;
 
+typedef struct {
+  UINT32                   Identifier;  //signature ('P', 'U', 'B', 'K')
+  UINT16                   KeySize;     //Length of Public Key
+  UINT8                    KeyType;     //RSA or ECC
+  UINT8                    Rsvd;
+  UINT8                    KeyData[0];  //Pubic key data(RSA_PUB_KEY) with KeySize bytes
+} PUB_KEY;
+
+typedef struct {
+  UINT32                   Identifier; //signature Identifier('S', 'I', 'G', 'N')
+  UINT16                   SigSize;    //Length of signature 2K and 3K in bytes
+  UINT8                    SigType;    //PKCSv1.5 or RSA-PSS or ECC
+  UINT8                    HashAlg;    //Hash Alg for signingh SHA256, SHA384
+  UINT8                    Signature[0];  //Signature length defined by SigSize bytes
+} SIGNATURE;
+
+
+#define RSA_SIGNATURE_SIZE         (RSA2048NUMBYTES + sizeof(SIGNATURE))
+#define RSA_KEYSIZE_SIZE           (RSA_MOD_SIZE + RSA_E_SIZE + sizeof(PUB_KEY))
+#define RSA_SIGNATURE_AND_KEY_SIZE  (RSA_SIGNATURE_SIZE + RSA_KEYSIZE_SIZE)
```

Comment:
Should avoid hard coded length calculation.  Using dynamic calculation instead.

Reply:
Dynamic calculation would be done as subsquently with "Hardcode removal in Ipp Crypto/Secureboot lib"